### PR TITLE
Move `api_server` dependencies from serve.txt to runtime.txt

### PR DESCRIPTION
**Pull request by lvhan028**
## Motivation

To simplify the installation section in README, which will be modified later

### CODE REVIEW

**Review by AllentDan — APPROVED** [review]

File: requirements/serve.txt [hunk 1]
```diff
@@ -1,6 +1,2 @@
-fastapi
 gradio<4.0.0
```
Comment (RunningLeon) [inline requirements/serve.txt:1]: could we add protobuf as required here 
https://github.com/InternLM/lmdeploy/blob/cc316a3e582e4a2d66d6e72f5fb41bb36a2f24ea/lmdeploy/serve/turbomind/chatbot.py#L12

Comment (lvhan028) [reply requirements/serve.txt:1]: Sure. Is there any constraints on its version?

Comment (RunningLeon) [reply requirements/serve.txt:1]: No version restriction.

Comment (lvhan028) [reply requirements/serve.txt:1]: OK. `protobuf` is added to `serve.txt`

**Pull request merged**